src/common/types/api.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import type { ElementOrigin } from '../../elements/common/flowTypes';
+
+type FetchOptions = {
+    fields?: Array<string>,
+    forceFetch?: boolean,
+    noPagination?: boolean,
+    refreshCache?: boolean,
+};
+
+type ErrorResponseData = {
+    code: string,
+    context_info: Object,
+    help_url: string,
+    message: string,
+    request_id: string,
+    status: number,
+    type: 'error',
+};
+
+type ElementsXhrError = $AxiosError<any> | ErrorResponseData;
+
+type ElementsError = {
+    code: string,
+    context_info: Object,
+    message: string,
+    origin: ElementOrigin,
+    type: 'error',
+};
+
+type ErrorContextProps = {
+    onError: (error: ElementsXhrError | Error, code: string, contextInfo?: Object, origin?: ElementOrigin) => void,
+};
+
+type ElementsErrorCallback = (e: ElementsXhrError, code: string, contextInfo?: Object) => void;
+
+export type {
```

Comment:
Curious if its better to `export` on each line? if possible, that's a small amount less code to maintain.

Reply:
Yeah, I have a slight preference to that for type files personally, but I think we collectively landed on this style instead. I suppose an argument could be made to do these type files the same as other JS files...?

Reply:
If there are a lot of internal types that aren't exported, having them on the bottom seems a bit easier for identifying what is actually exported. But when everything is exported, seems like the opposite. 🤔